### PR TITLE
[MetaSchedule] Tuning Script Upgrade

### PR DESCRIPTION
**Pull request by zxybazh**
This PR introduces changes to upgrade the onnx / relay tuning script:
- [x] support int8 / uint8 as testing input for onnx and relay
- [x] integration test for int8 / uint8
- [x] refactor functions to clearly display per layer and e2e performance.
- [x] support using VirtualMachine in addtion to GraphModule as a script option.
- [x] integration test to check VM support

### CODE REVIEW

**Comment by junrushao1994** [comment]
let's rebase against the lastest HEAD

**Comment by zxybazh** [comment]
Rebase and ready for review!

File: python/tvm/meta_schedule/testing/custom_builder_runner.py [hunk 1]
```diff
@@ -158,15 +160,22 @@ def run_module_via_rpc(
...
+            rt_mod = session.load_module(filename)
+            if backend == "vm":
+                rt_mod = session.get_function("runtime.Load_Executable")(code, rt_mod)
+            dev = session.device(dev_type=dev_type, dev_id=0)
+            nd_args = {k: ndarray.array(v, dev) for k, v in args.items()}
+            return continuation(rt_mod, dev, nd_args)
+    except Exception as exc:  # pylint: disable=broad-except
+        print(
+            f"Run module {continuation.__name__} via RPC failed, exception: {exc}",
+        )
```
Comment (junrushao1994) [inline python/tvm/meta_schedule/testing/custom_builder_runner.py:181]: quick question: why do we need to silence the exceptions here?

File: python/tvm/meta_schedule/testing/utils.py [hunk 2]
```diff
@@ -77,3 +81,145 @@ def apply_fixed_schedules(
...
+        return np.random.uniform(size=input_shape).astype(input_dtype)
+    if input_dtype in ["uint8", "int8"]:
+        return np.random.randint(
+            low=0,
+            high=127,
+            size=input_shape,
+            dtype="int32",  # TODO(zxybazh): fix the datatype when int8 / uint8 is supported better
+        )
+    if input_dtype in ["int32", "int64"]:
+        return np.random.randint(low=0, high=10000, size=input_shape, dtype=input_dtype)
```
Comment (junrushao1994) [inline python/tvm/meta_schedule/testing/utils.py:111]: If it's an indexing table used in embedding lookups, then `low` and `high` here probably indicate the size of the embedding table. If not taken care of, could cause some buffer access overflow. I don't have good ideas of how to properly handling this, but let's add a warning here

File: python/tvm/meta_schedule/testing/utils.py [hunk 3]
```diff
@@ -77,3 +81,145 @@ def apply_fixed_schedules(
...
+    raise ValueError("Unsupported input datatype!")
+
+
+def f_timer(backend: str) -> Callable:
```
Comment (junrushao1994) [inline python/tvm/meta_schedule/testing/utils.py:115]: let's find a better name. how about, `create_timer_func`

File: python/tvm/meta_schedule/testing/utils.py [hunk 4]
```diff
@@ -77,3 +81,145 @@ def apply_fixed_schedules(
...
+        print("-------------------------------")
+        print(f"    Min (ms) : {min(results)}")
+        print(f"    Max (ms) : {max(results)}")
+        print(f" Median (ms) : {median(results)}")
+        print(f"Average (ms) : {sum(results) / len(results)}")
+
+    return f_timer_func
+
+
+def f_per_layer(graph: str) -> Callable:
```
Comment (junrushao1994) [inline python/tvm/meta_schedule/testing/utils.py:174]: ditto. how about `create_per_layer_timer_func`

File: python/tvm/meta_schedule/testing/custom_builder_runner.py (hunk 1 above)
Comment (zxybazh) [reply python/tvm/meta_schedule/testing/custom_builder_runner.py:181]: Yeah it's too broad, I'll move that into continuation funciton.

File: python/tvm/meta_schedule/testing/utils.py (hunk 2 above)
Comment (zxybazh) [reply python/tvm/meta_schedule/testing/utils.py:111]: That's right, for `gpt-2` and `bert` the embedding table is not the same. In that case we need user input, so I'll just add the low and high as argument in this function and throw a warning.

**Review by junrushao1994 — APPROVED** [review]
LGTM

**Review by junrushao1994 — APPROVED** [review]
LGTM

**Pull request merged**